cli/pkg/update/update.go

```diff
@@ -0,0 +1,209 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/hashicorp/go-version"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	StateFilename       = "state.yaml"
+	VersionInfoStateKey = "version_info"
+	Addr                = "https://api.github.com/repos/rilldata/rill-developer/releases/latest"
+)
+
+type UpdateInfo struct {
+	Update      bool
+	Message     string
+	ReleaseInfo *ReleaseInfo
+}
+
+// ReleaseInfo stores information about a release
+type ReleaseInfo struct {
+	Version     string    `json:"tag_name"`
+	URL         string    `json:"html_url"`
+	PublishedAt time.Time `json:"published_at"`
+}
```

Comment:
Make it `githubReleaseInfo` and define it right above the function that returns it